src/sys/signal.rs

```diff
@@ -722,6 +735,25 @@ mod tests {
         assert_eq!(action_ign.handler(), SigHandler::SigIgn);
     }
 
+    #[test]
+    fn test_procalive() {
+        use sys::wait::waitpid;
+        use unistd::fork;
+        use unistd::ForkResult::*;
+
+        match fork() {
```

Comment:
Calling `fork` conflicts with other tests that call `wait`.  So you must grab the `FORK_MTX` at the top of the test.